source/ada/lsp-ada_documents.ads

```diff
@@ -94,6 +95,18 @@ package LSP.Ada_Documents is
       return Libadalang.Analysis.Ada_Node;
    --  Get Libadalang Node for given position in the document.
 
+   function Get_Root_Node
```

Comment:
May I suggest to not expose the root node, but move folding population code into the `Ada_Documents` instead? It would make Ada_Handler a bit shorter.